### PR TITLE
refactor: batchwise indexing

### PR DESCRIPTION
**Pull request by joschkabraun**
Fixes:
- removes necessity to index in batches
- specifies gateway instance to C4 to index larger amount of documents without OOM
- enables passing of AWS credentials via CLI (3rd PR from me now doing this lol)

Note, this branches off https://github.com/jina-ai/now/pull/1107.

---

- [x] This PR references an open issue
- [x] Added a test
- [x] Added a screenshot of a successful customer deployment
Screenshot 2023-02-27 at 12 49 59

### CODE REVIEW

File: tests/unit/test_dataclass.py [hunk 1]
```diff
@@ -117,11 +122,8 @@ def test_create_annotations_and_class_attributes(
             ['price', 'description'],
             {'price': float, 'description': str},
             {
-                'image_0': S3Object,
-                'text_0': S3Object,
-                'filter_0': S3Object,
-                'filter_1': S3Object,
-                'json_s3': S3Object,
```
Comment (Tanguyabel) [inline tests/unit/test_dataclass.py:124]: is there a specific reason why you remove those?

**Review by Tanguyabel — APPROVED** [review]
little comment about the test, otherwise lgtm!

File: tests/unit/test_dataclass.py (hunk 1 above)
Comment (joschkabraun) [reply tests/unit/test_dataclass.py:124]: it was outdated, these aren't indexed fields and shouldn't appear in the output of the mm doc class

Comment (Tanguyabel) [reply tests/unit/test_dataclass.py:124]: ok, good then!

**Pull request merged**